### PR TITLE
Fix crossdock and Travis issues

### LINKED ISSUE #1028: Crossdock and Travis are buggy
## Requirement - what kind of business use case are you trying to solve?

~~Ever since #815, the crossdock tests have been broken.~~ This should have been caught immediately, but an error-handling issues in `.travis.yml` allowed the failed test to be ignored.

## Problem - what in Jaeger blocks you from solving the requirement?

The existing crossdock config and Travis CI setup have multiple issues:

- DNS config is not set up using `dns_search: .`, which sporadically causes errors when the a service never initializes, when in fact it has initialized, but the DNS config fails.
- ~~Command format (from #815) does not use the executable as the first argument, which is invalid.~~
- Using locally tagged images without specifying `:latest` causes build failures associated with outdated versions of the image.
- When using `if`/`then` in `.travis.yml`, use `test || exit 1` to guarantee test succeeds. Otherwise, the `then` command succeeds regardless of exit code from `test`.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Fix the above issues.

### PR DESCRIPTION
**Pull request by isaachier**
Signed-off-by: Isaac Hier <ihier@uber.com>

## Which problem is this PR solving?
- Fixes #1028.

## Short description of the changes
- Fixes problems listed in #1028.

### CODE REVIEW

File: cmd/collector/Dockerfile [hunk 1]
```diff
@@ -7,4 +7,4 @@ COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certifica
 
 EXPOSE 14267
 COPY collector-linux /go/bin/
-ENTRYPOINT ["/go/bin/collector-linux"]
+ENTRYPOINT ["/go/bin/collector-linux", "--cassandra.keyspace=jaeger_v1_dc1", "--cassandra.servers=cassandra", "--collector.zipkin.http-port=9411"]
```
Comment (black-adder) [inline cmd/collector/Dockerfile]: We might need another solution. I pulled this branch and tried to build and run the collector container with non-cassandra memory store:
```
wjang:jaeger won$ docker run -e SPAN_STORAGE_TYPE=memory jaegertracing/jaeger-collector:latest
unknown flag: --cassandra.keyspace
Error: unknown flag: --cassandra.keyspace
```

File: crossdock/docker-compose.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ version: '2'
 services:
     crossdock:
         image: crossdock/crossdock
+        dns_search: .  # Ensures unified DNS config.
```
Comment (vprithvi) [inline crossdock/docker-compose.yml]: Could you elaborate on how this ensures unified DNS config?
Isn't  `.` an invalid IP Address or host name?

File: Makefile [hunk 3]
```diff
@@ -262,10 +262,6 @@ build-crossdock-ui-placeholder:
 build-crossdock: build-crossdock-ui-placeholder docker-no-ui
 	make crossdock
 
-.PHONY: build-crossdock-fresh
```
[the remark this answers is not in the document]
Comment (isaachier) [reply Makefile]: If it was, it wasn't updated when `crossdock-fresh` was deleted from `crossdock/rules.mk`. That might be my own fault.

File: crossdock/docker-compose.yml (hunk 2 above)
Comment (isaachier) [reply crossdock/docker-compose.yml]: I wish I understood this. I can try to get some more info and will post here.

File: cmd/collector/Dockerfile (hunk 1 above)
Comment (isaachier) [reply cmd/collector/Dockerfile]: OK I see the problem. The solution is probably using the original `command` override but adding the program as the first argument.

File: crossdock/docker-compose.yml (hunk 2 above)
Comment (isaachier) [reply crossdock/docker-compose.yml]: OK have a better solution and explained in comment.

Comment (vprithvi) [reply crossdock/docker-compose.yml]: What is the problem being solved?

Comment (isaachier) [reply crossdock/docker-compose.yml]: Well using `8.8.8.8` showed me that using a real DNS search host can mess up the test. The docker-compose hostnames won't be resolved correctly. The motive for using `dns_search: .` is described [here](https://github.com/crossdock/crossdock/pull/64). I will add a link in the comment.

**Comment by isaachier** [comment]
I'm gonna close this. Looks like I was dead wrong about the `if`/`then` thing too. Just my MacBook being buggy. If anyone else experiences the DNS issue, I can fix that then.

**Pull request closed without merging**